src/hint_processor/hint_processor_def.zig

```diff
@@ -53,6 +94,42 @@ pub const HintReference = struct {
     }
 };
 
+/// Takes a mapping from reference name to reference id, normalizes the reference name, and maps the normalized reference name to its denoted HintReference from a reference array.
+///
+/// Arguments:
+///   - `allocator`: The allocator that shares scope with the operating HintProcessor.
+///   - `reference_ids`: A mapping from reference name to reference id
+///   - `references`: An array of HintReferences, as indexed by reference id
+///
+/// # Returns
+/// A mapping of normalized reference name to HintReference, if successful
+///
+/// Errors
+/// - If there is no corresponding reference in `references` array
+pub fn getIdsData(allocator: Allocator, reference_ids: StringHashMap(usize), references: []const HintReference) !StringHashMap(HintReference) {
+    var ids_data = StringHashMap(HintReference).init(allocator);
+    errdefer ids_data.deinit();
+
+    var ref_id_it = reference_ids.iterator();
+    while (ref_id_it.next()) |ref_id_entry| {
+        const path  = ref_id_entry.key_ptr.*;
+        const ref_id = ref_id_entry.value_ptr.*;
+
+        var name_iterator = std.mem.splitBackwards(u8, path, @as([]const u8, "."));
+        
+        const name = name_iterator.next() orelse return CairoVMError.Unexpected;
+
+        if (ref_id >= references.len) return CairoVMError.Unexpected;
```

Comment:
I think you should put this condition at the beginning of the loop to avoid doing the rest of the calculations if this one is not satisfied what do you think?